### PR TITLE
Add overloaded name to prim operators

### PR DESCRIPTION
**Pull request by iseeyuan**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#34280 Add overloaded name to prim operators**

To have prim ops searchable for lite interpreter, overloaded names need to be added for the operators with the same name but different schema. For example, aten::add in register_prim_ops.cpp. The difference is a combination of args and output type.
`"aten::add(str a, str b) ->str"`
`"aten::add(int a, int b) ->int"`
`"aten::add(float a, float b) ->float"`
`"aten::add(int a, float b) ->float"`
`"aten::add(float a, int b) ->float"`
`"aten::add(Scalar a, Scalar b) ->Scalar"`

Solution: 
Use the argument type and/or output type (the same to the existing overloaded names). The overloaded name should be minimum as long as the operators can be differentiated. For other operators please look into the source code change for details. 

`"aten::add.str(str a, str b) ->str"`
`"aten::add.int(int a, int b) ->int"`
`"aten::add.float(float a, float b) ->float"`
`"aten::add.int_float(int a, float b) ->float"`
`"aten::add.float_int(float a, int b) ->float"`
`"aten::add.Scalar_Scalar(Scalar a, Scalar b) ->Scalar"`

Differential Revision: [D20456997](https://our.internmc.facebook.com/intern/diff/D20456997)

### CODE REVIEW

**Comment by eellison** [comment]
Before we land this, can you remove the specialized list ops ? There is no reason for them to exist anymore, and we won't be able to remove them after this is landed without breaking BC.

File: torch/csrc/jit/runtime/register_prim_ops.cpp [hunk 1]
```diff
@@ -1231,7 +1231,7 @@ RegisterOperators logging_operators(
 // define implementations for primitive number ops
 #define DEFINE_GENERIC_OP(aten_op, int_op, float_op, int_result, float_result) \
   Operator(                                                                    \
-      #aten_op "(int a, int b) -> " #int_result,                               \
+      #aten_op ".int(int a, int b) -> " #int_result,                               \
```
Comment (smessmer) [inline torch/csrc/jit/runtime/register_prim_ops.cpp]: nit: line endings moved, here and in other places. Maybe run clang-format?

**Review by smessmer — APPROVED** [review]
Please also address @eellison's comment before landing

**Comment by iseeyuan** [comment]
> Before we land this, can you remove the specialized list ops ? There is no reason for them to exist anymore, and we won't be able to remove them after this is landed without breaking BC.

@eellison Could you specify from the code where are the specialized list ops?

**Comment by zdevito** [comment]
@iseeyuan the macros like CREATE_MUTABLE_LIST_OPS that were modified in the PR define versions of an op for Tensor[], int[], float[], bool[], and t[]. Only the t[] versions should remain since we no longer have specialized versions of lists for each op. We need to fix this before we generate overload names because we do not want to permanently preserve these ops in the bytecode.

**Comment by iseeyuan** [comment]
> @iseeyuan the macros like CREATE_MUTABLE_LIST_OPS that were modified in the PR define versions of an op for Tensor[], int[], float[], bool[], and t[]. Only the t[] versions should remain since we no longer have specialized versions of lists for each op. We need to fix this before we generate overload names because we do not want to permanently preserve these ops in the bytecode.

Thanks! @eellison and I were looking at this. I'll remove the ops before landing this PR.

**Pull request closed without merging**